tests/test_flows.py

```diff
@@ -401,18 +401,86 @@ def test_channel_selection(self):
         self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_PREFERENCE_QUERY_MESSAGE")
         self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_PREFERENCE_QUERY_MESSAGE")
 
-        self.debug("Send channel selection request")
-        self.gateway_ucc.dev_send_1905(self.mac_repeater1, 0x8006)
-        time.sleep(1)
-        self.debug("Confirming channel selection request has been received on agent")
+        # Send empty channel selection request
+        self.debug("Send empty channel selection request")
```

Comment:
A comment that repeats the log message is redundant.

However, this is not what happens (or should happen). When dev_send_1905 without TLVs is done, the controller should add the TLVs itself. Cfr. step 5 of 5.5.1: "MCUT to send a Channel Selection Request message to CTT Agent1 for each Channel Preference TLV received in the Channel Preference Report message. (MCUT is expected to
generate the Channel Preference TLV)"

So the problem is actually in the controller, in that it doesn't send a channel preference TLV while it should do so. Probably it's because the agent sends an empty channel preference TLV.

So the problem is in handle_cmdu_1905_channel_preference_report() which doesn't correctly fill the certification buffer.

That said, we should obviously still have this test with a completely empty channel selection request, because it does expose a corner case with a bug. A trick to achieve that is:

```
self.gateway_ucc.dev_send_1905(self.mac_repeater1, 0x8006, tlv(0x00, 0x0000, "")
```

That adds an EOM TLV which would otherwise anyway be added automatically, but to the CAPI parser it looks like a TLV was provided.